waitress/channel.py

```diff
@@ -344,6 +369,9 @@ def service(self):
                 else:
                     task = self.task_class(self, request)
                 try:
+                    # before processing a new request, ensure there is not
+                    # too much data in the outbufs waiting to be flushed
+                    self.flush_outbufs_below_high_watermark(lock=True)
```

Comment:
I intentionally placed this before the request starts instead of down below in the `else` because it's possible to exhaust current requests, exit the loop, and receive a new request right away while still sending data from the previous loop. Checking at the start of the request will ensure we do not begin the request until the previous request's data is sufficiently out the door.

That being said, `readable()` currently returns `False` while `self.any_outbuf_has_data()` which means currently it would be slightly more efficient to put it below on line 416 and only do it if there is another request to process.

Reply:
I decided ultimately to move the check down below due to the reasons stated above.